src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_IIntellisenseBuildTarget.cs

```diff
@@ -0,0 +1,134 @@
+            }
+        }
+
+        private void UpdateHostDiagnostics(bool succeeded, string reason)
+        {
+            if (!succeeded)
+            {
+                // report intellisense build failure to error list
+                this.HostDiagnosticUpdateSource?.UpdateDiagnosticsForProject(
+                    _id, s_diagnosticKey, SpecializedCollections.SingletonEnumerable(CreateIntellisenseBuildFailureDiagnostic(reason)));
+            }
+            else
+            {
+                // clear intellisense build failure diagnostic from error list.
+                this.HostDiagnosticUpdateSource?.ClearDiagnosticsForProject(_id, s_diagnosticKey);
+            }
+        }
+
+        private DiagnosticData CreateIntellisenseBuildFailureDiagnostic(string reason)
+        {
+            // log intellisense build failure
+            Logger.Log(FunctionId.IntellisenseBuild_Failed, KeyValueLogMessage.Create(m => m["Reason"] = reason ?? string.Empty));
+
+            return new DiagnosticData(
+                IDEDiagnosticIds.IntellisenseBuildFailedDiagnosticId,
+                FeaturesResources.ErrorCategory,
+                ServicesVSResources.IntellisenseBuildFailedMessage,
+                ServicesVSResources.ResourceManager.GetString(nameof(ServicesVSResources.IntellisenseBuildFailedMessage), CodeAnalysis.Diagnostics.Extensions.s_USCultureInfo),
+                DiagnosticSeverity.Warning,
+                isEnabledByDefault: true,
+                warningLevel: 0,
+                workspace: Workspace,
+                projectId: Id,
+                title: ServicesVSResources.IntellisenseBuildFailedTitle,
+                description: GetDescription(reason));
+        }
+
+        private string GetDescription(string reason)
+        {
+            var logFilePath = GetLogFile();
+            var logFileDescription = string.Format(
+                    logFilePath == null ? ServicesVSResources.IntellisenseBuildFailedDescription : ServicesVSResources.IntellisenseBuildFailedDescriptionWithLog,
+                    logFilePath ?? GeneralLogFile);
+
+            if (string.IsNullOrWhiteSpace(reason))
+            {
+                return logFileDescription;
+            }
+
+            return string.Join(Environment.NewLine, logFileDescription, string.Empty, ServicesVSResources.IntellisenseBuildFailedDescriptionExtra, reason);
+        }
+
+        private string GetLogFile()
+        {
+            var manager = BuildManager.DefaultBuildManager;
+            if (manager == null)
+            {
+                return null;
+            }
+
+            var buildParameters = s_fieldInfo.GetValue(manager) as BuildParameters;
+            if (buildParameters == null)
+            {
+                return null;
+            }
+
+            var fileLogger = buildParameters.Loggers?.OfType<FileLogger>()?.FirstOrDefault();
```

Comment:
@jasonmalinowski well, the good blog post still can't let you know the exact log file name. that is the problem. only way is you go through all designtime.log files in the directory, and open one by one or do some kind of file in directory search to find what you want. all just extra hurdle for users to get to information they need.